examples/authentication/server/src/auth_handler.rs

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! gRPC service implementation for handling authentication requests from clients.
+
+use proto::{
+    authentication_server::{Authentication, AuthenticationServer},
+    AuthParameters, IdentityTokenRequest, IdentityTokenResponse,
+};
+use tonic::{Request, Response, Status};
+
+pub mod proto {
+    tonic::include_proto!("oak.examples.authentication");
+}
+
+use crate::token_exchanger;
+
+pub fn build_service(
+    client_id: &str,
+    client_secret: &str,
+) -> AuthenticationServer<AuthenticationHandler> {
+    AuthenticationServer::new(AuthenticationHandler::new(client_id, client_secret))
+}
+
+#[derive(Default)]
+pub struct AuthenticationHandler {
+    client_id: String,
+    client_secret: String,
+    /* TODO(#922): Add a state storage mechanism to be able to match a client request to the
+     * nonce and code challenge. */
+}
+
+impl AuthenticationHandler {
+    pub fn new(client_id: &str, client_secret: &str) -> AuthenticationHandler {
+        AuthenticationHandler {
+            client_id: client_id.to_owned(),
+            client_secret: client_secret.to_owned(),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl Authentication for AuthenticationHandler {
+    /// Gets the authentication parameters needed by the client to send authorisation token
```

Comment:
```suggestion
    /// Gets the authentication parameters needed by the client to send authorisation code
```
Is an `authorisation token` the same thing as an `authorisation code`?